WooCommerce/Classes/Styles/UIColor+SemanticColors.swift

```diff
@@ -0,0 +1,318 @@
+    ///
+    static var warning: UIColor {
+        return UIColor(light: muriel(color: .yellow, .shade50),
+                        dark: muriel(color: .yellow, .shade30))
+    }
+
+    /// Blue. Blue-50 (< iOS 13 and Light Mode) and Blue-30 (Dark Mode)
+    ///
+    static var blue: UIColor {
+        return UIColor(light: muriel(color: .blue, .shade50),
+                        dark: muriel(color: .blue, .shade30))
+    }
+
+    /// Orange. Orange-50 (< iOS 13 and Light Mode) and Orange-30 (Dark Mode)
+    ///
+    static var orange: UIColor {
+        return UIColor(light: muriel(color: .orange, .shade50),
+                        dark: muriel(color: .orange, .shade30))
+    }
+}
+
+
+// MARK: - Text Colors.
+extension UIColor {
+    /// Text. Gray-80 (< iOS 13) and `UIColor.label` (> iOS 13)
+    ///
+    static var text: UIColor {
+        if #available(iOS 13, *) {
+            return .label
+        }
+
+        return .gray(.shade80)
+    }
+
+    /// Text Subtle. Gray-50 (< iOS 13) and `UIColor.secondaryLabel` (> iOS 13)
+    ///
+    static var textSubtle: UIColor {
+        if #available(iOS 13, *) {
+            return .secondaryLabel
+        }
+
+        return .gray(.shade50)
+    }
+
+    /// Text Tertiary. Gray-20 (< iOS 13) and `UIColor.tertiaryLabel` (> iOS 13)
+    ///
+    static var textTertiary: UIColor {
+        if #available(iOS 13, *) {
+            return .tertiaryLabel
+        }
+
+        return .gray(.shade20)
+    }
+
+    /// Text Quaternary. Gray-10 (< iOS 13) and `UIColor.quaternaryLabel` (> iOS 13)
+    ///
+    static var textQuaternary: UIColor {
+        if #available(iOS 13, *) {
+            return .quaternaryLabel
+        }
+
+        return .gray(.shade10)
+    }
+
+    /// Text Inverted. White(< iOS 13 and Light Mode) and Gray-90 (Dark Mode)
+    ///
+    static var textInverted: UIColor {
+        return UIColor(light: .white,
+                       dark: muriel(color: .gray, .shade90))
+    }
+
+    /// Text Placeholder. Gray-30 (< iOS 13) and `UIColor.placeholderText` (> iOS 13)
+    ///
+    static var textPlaceholder: UIColor {
+        if #available(iOS 13, *) {
+            return .placeholderText
+        }
+
+        return .gray(.shade30)
+    }
+}
+
+
+// MARK: - UI elements.
+extension UIColor {
+    /// Basic Background. White (< iOS 13) and `UIColor.systemBackground` (> iOS 13)
+    ///
+    static var basicBackground: UIColor {
+        if #available(iOS 13, *) {
+            return .systemBackground
+        }
+
+        return .white
+    }
+
+    /// App Bar. WooCommercePurple-60 (< iOS 13 and Light Mode) and `UIColor.systemThickMaterial` (Dark Mode)
+    ///
+    static var appBar: UIColor {
+        if #available(iOS 13, *) {
+            return UIColor(light: muriel(color: .wooCommercePurple, .shade60),
+                           dark: .systemBackground)
+        }
+
+
+        return muriel(color: .wooCommercePurple, .shade60)
```

Comment:
could the function just return `UIColor(light: muriel(color: .wooCommercePurple, .shade60), dark: .systemBackground)` since the light color is returned for < iOS 13?

Reply:
Right! Nice catch, thanks!

Reply:
Oh, no, wait! It would not compile, `.systemBackground` is iOS 13 only